airflow/api_connexion/schemas/variable_schema.py

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from marshmallow import Schema, fields
+
+
+class VariableSchema(Schema):
+    """ Variable Schema """
+    key = fields.String(required=True)
+    value = fields.String(attribute="val", required=True)
+
+
+class VariableCollectionSchema(Schema):
+    """ Variable Collection Schema """
+    variables = fields.List(fields.Nested(VariableSchema))
+    total_entries = fields.Int()
+
```

Comment:
I would do this schema as follow:
```
class VariableCollectionItemSchema(SQLAlchemySchema):
    """ A schema for variable collection item """
    class Meta:
        """ Meta """
        model = Variable
    key = auto_field(required=True)


class VariableSchema(VariableCollectionItemSchema):
    """ Variable Schema """
    val = auto_field(dump_to='value', load_from='value', required=True)

class VariableCollectionSchema(Schema):
    """ Variable Collection Schema """
    variables = fields.List(fields.Nested(VariableSchema))
    total_entries = fields.Int()

```
The spec has this to say about VariableCollectionItemSchema : `Divided into two schemas for sensitive data protection

Reply:
good catch